service/awsconfig/drainer_framework.go

```diff
@@ -1,16 +1,164 @@
+	GuestUpdateEnabled bool
+	ProjectName        string
+}
+
+type DrainerFrameworkConfigAWS struct {
+	AccessKeyID     string
+	AccessKeySecret string
+	Region          string
+	SessionToken    string
 }
 
 func NewDrainerFramework(config DrainerFrameworkConfig) (*framework.Framework, error) {
-	return nil, nil
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.K8sExtClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sExtClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.AWS.AccessKeyID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWS.AccessKeyID must not be empty", config)
+	}
+	if config.AWS.AccessKeySecret == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWS.AccessKeySecret must not be empty", config)
+	}
+	if config.AWS.Region == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWS.Region must not be empty", config)
+	}
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ProjectName must not be empty", config)
+	}
+
+	var err error
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.DefaultConfig()
+
+		c.K8sExtClient = config.K8sExtClient
+		c.Logger = config.Logger
+
+		crdClient, err = k8scrdclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resourceRouter, err := newDrainerResourceRouter(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
+
+		c.Watcher = config.G8sClient.ProviderV1alpha1().AWSConfigs("")
+
+		c.ResyncPeriod = 30 * time.Second
+
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var crdFramework *framework.Framework
+	{
+		c := framework.Config{
+			CRD:            v1alpha1.NewAWSConfigCRD(),
+			CRDClient:      crdClient,
+			Informer:       newInformer,
+			Logger:         config.Logger,
+			ResourceRouter: resourceRouter,
+		}
+
+		crdFramework, err = framework.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	return crdFramework, nil
 }
 
 func newDrainerResourceRouter(config DrainerFrameworkConfig) (*framework.ResourceRouter, error) {
-	return nil, nil
+	var err error
+
+	var awsClients awsclient.Clients
+	{
+		c := awsclient.Config{
+			AccessKeyID:     config.AWS.AccessKeyID,
+			AccessKeySecret: config.AWS.AccessKeySecret,
+			SessionToken:    config.AWS.SessionToken,
+			Region:          config.AWS.Region,
+		}
+
+		awsClients = awsclient.NewClients(c)
+	}
+
+	var v7ResourceSet *framework.ResourceSet
+	{
+		c := v7.DrainerResourceSetConfig{
```

Comment:
One weird thing I am not sure about yet is that when we have multiple frameworks we have to bump this in different files even though nothing changes in certain files essentially, but the the imports. We maybe have to revisit this and extend the concepts operatorkit provides. FYI @kopiczko.

Reply:
You mean something changes in `Cluster` but nothing in `Drainer` and still there is version bump in `Drainer`?

Reply:
Yes. This is odd. Tooling could do this but we are far away from this kind of tooling IMO. This here makes me question our approach but anything else I come up with is just worse.